### DISCUSSION TITLE
Exception in callback H11Protocol.timeout_keep_alive_handler()

### DISCUSSION THREAD

**Discussion by sboudouk** [Kludex/uvicorn#1984] [Potential Issue]
Hello,

I'm using gunicorn with FastAPI & Uvicorn workers (and not h11uvicorn workers), and from time to time I get this error that I'm unable to reproduce:

[ERROR] Exception in callback H11Protocol.timeout_keep_alive_handler()

I'm not hoping for a solution here but more of hints if you guys have any idea what can produce this since I found little to no help by searching this error earlier.

Just for info, i'm running uvicorn 0.20 and plan to update it to latest 0.22 this week.

Thanks for your help 👍

**Comment by Kludex** [comment]
Can you show the whole logs, and an [MRE](https://stackoverflow.com/help/minimal-reproducible-example)?

**Reply by sboudouk** [reply]
I can surely show the whole logs, but i can't reproduce it myself unfortunately otherwise I would have opened an issue or fixed it with a PR

Here is the whole logs:

```
Exception in callback H11Protocol.timeout_keep_alive_handler() handle: <TimerHandle when=16036046.86233932 H11Protocol.timeout_keep_alive_handler()> [] Traceback (most recent call last): []   File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run []     self._context.run(self._callback, *self._args) []   File "/opt/pysetup/.venv/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 362, in timeout_keep_alive_handler []     self.conn.send(event) []   File "/opt/pysetup/.venv/lib/python3.9/site-packages/h11/_connection.py", line 512, in send []     data_list = self.send_with_data_passthrough(event) []   File "/opt/pysetup/.venv/lib/python3.9/site-packages/h11/_connection.py", line 537, in send_with_data_passthrough []     self._process_event(self.our_role, event) []   File "/opt/pysetup/.venv/lib/python3.9/site-packages/h11/_connection.py", line 272, in _process_event []     self._cstate.process_event(role, type(event), server_switch_event) []   File "/opt/pysetup/.venv/lib/python3.9/site-packages/h11/_state.py", line 293, in process_event []     self._fire_event_triggered_transitions(role, _event_type) []   File "/opt/pysetup/.venv/lib/python3.9/site-packages/h11/_state.py", line 311, in _fire_event_triggered_transitions []     raise LocalProtocolError( [] h11._util.LocalProtocolError: can't handle event type ConnectionClosed when role=SERVER and state=SEND_RESPONSE  
```

+ I'm also using FastAPI alongwith gunicorn & uvicorn workers

**Comment by hackrole** [comment]
Same Issues, I can provide more exception message.

![image](https://github.com/encode/uvicorn/assets/2100811/3b751fb9-7fac-401c-9df4-5b7243bd488b)

the keep-alive is called exactly after 5 seconds of the requests start. the following log is from our middleware implementation.
![image](https://github.com/encode/uvicorn/assets/2100811/a0620729-ccc7-453b-a16e-9e3293f7ec95)


I use uvicorn 0.29.0
```
$ pip freeze | ag '(uvic|h11)'
h11==0.14.0
uvicorn==0.29.0
```

And this error will cause next error like below.
![image](https://github.com/encode/uvicorn/assets/2100811/37c053c1-1a82-4d96-95b4-dd7564e0e6cd)

**Reply by hackrole** [reply]
And our API is a proxy to chatGPT, it return SSE streamResponse, and usually token about 20 seconds or more to finish the response.
